pandas/core/reshape/merge.py

```diff
@@ -837,7 +837,7 @@ def _maybe_add_join_keys(self, result, left_indexer, right_indexer):
                 # make sure to just use the right values
                 mask = left_indexer == -1
                 if mask.all():
-                    key_col = rvals
+                    key_col = Index(rvals)
```

Comment:
Do we need to propagate a `name` here?

Reply:
No, the name is handled in line 846 and 856.

But we could save the index call in line 856 if this is performance relevant?